tests/snmp/test_snmp_interfaces.py

```diff
@@ -10,6 +12,85 @@
 
 logger = logging.getLogger(__name__)
 
+SAI_PORT_STAT_IF_IN_ERRORS = 'SAI_PORT_STAT_IF_IN_ERRORS'
+SAI_PORT_STAT_IF_OUT_ERRORS = 'SAI_PORT_STAT_IF_OUT_ERRORS'
+SAI_PORT_STAT_IF_IN_DISCARDS = 'SAI_PORT_STAT_IF_IN_DISCARDS'
+SAI_PORT_STAT_IF_OUT_DISCARDS = 'SAI_PORT_STAT_IF_OUT_DISCARDS'
+SAI_ROUTER_INTERFACE_STAT_IN_ERROR_PACKETS = 'SAI_ROUTER_INTERFACE_STAT_IN_ERROR_PACKETS'
+SAI_ROUTER_INTERFACE_STAT_OUT_ERROR_PACKETS = 'SAI_ROUTER_INTERFACE_STAT_OUT_ERROR_PACKETS'
+
+COUNTERS_PORT_NAME_MAP = 'COUNTERS_PORT_NAME_MAP'
+COUNTERS_RIF_NAME_MAP = 'COUNTERS_RIF_NAME_MAP'
+COUNTER_VALUE = 5000
+
+
+@pytest.fixture()
+def disable_conterpoll(duthost):
+    """
+    Disable conterpoll for RIF and PORT and re-enable it when TC finished
+    :param duthost: DUT host object
+    :return: dict with data collected from DUT per each port
+    """
+    ConterpollHelper.disable_counterpoll(duthost, counter_type_list=['port', 'rif'])
+    yield
+    ConterpollHelper.enable_counterpoll(duthost, counter_type_list=['port', 'rif'])
+
+
+def get_interfaces(duthost, tbinfo):
+    """
+    Method to get interfaces for testing
+    :param duthost: DUT host object
+    :return: RIF interface name in case available in topo. If not - return Port Channel name and interface in Port
+     Channel
+    """
+    rif_counters = parse_rif_counters(duthost.command("show interfaces counters rif")["stdout_lines"])
+    for interface in rif_counters:
+        if 'Eth' in interface:
+            return interface, interface
+        else:
+            mg_facts = duthost.get_extended_minigraph_facts(tbinfo)
+            return mg_facts["minigraph_portchannels"][interface]['members'][0], interface
+
+
+def get_oid_for_interface(duthost, table_name, interface_name):
+    """
+    Method to get interface oid from Counters DB
+    :param duthost: DUT host object
+    :param table_name: table name
+    :param interface_name: interface name
+    :return: oid for specific interface
+    """
+    return duthost.command(f"docker exec -i database redis-cli --raw -n 2 HMGET "
+                           f"{table_name} {interface_name}")["stdout"]
```

Comment:
This will not work as expected for multi-asic devices since there are multiple database dockers. The test function could be modified to use enum_rand_one_per_hwsku_frontend_hostname, enum_asic_index to run the test for each frontend dut and each asic namespace. That namespace can be used in sonic-db-cli -n <namespace> to get the required mapping.